code/mob/living/critter/flockcritter_parent.dm

```diff
@@ -129,7 +128,8 @@
 	if(src.is_npc)
 		// tell the npc AI to go after the target
 		if(src.ai)
-			src.ai.rally(target)
+			var/datum/aiHolder/flock/flockai = ai
+			flockai.rally(target)
```

Comment:
why this weird change of var scope?

Reply:
rally is a proc only on flock AI and i moved 'ai' up into existing on mob to simplify checks in the mob_ai loop and reduce istype usage